src/lwes_event.c

```diff
@@ -698,10 +667,59 @@ lwes_event_add_headers
   return 0;
 }
 
+#define TYPED_BYTES_TO_EVENT_FIELD(typ,set_fail_ret,mar_fail_ret)    \
+  else if (tmp_byte == LWES_TYPE_##typ)                              \
+    {                                                                \
+      LWES_##typ var;                                                \
+      if (!unmarshall_##typ                                          \
+            (&var, bytes, num_bytes, &tmpOffset))                    \
+        { return mar_fail_ret; }                                     \
+      if (0 > lwes_event_set_##typ                                   \
+            (event, tmp_short_str, var))                             \
+        { return set_fail_ret; }                                     \
+    }                                                                \
+  else if (tmp_byte == LWES_TYPE_##typ##_ARRAY)                      \
+    {                                                                \
+      struct lwes_event_attribute* attr;                             \
+      attr = lwes_event_attribute_create(tmp_byte, NULL, 0);         \
+      if (!unmarshall_array_attribute                                \
+            (attr, bytes, num_bytes, &tmpOffset))                    \
+        {                                                            \
+          lwes_event_attribute_destroy(attr);                        \
+          return mar_fail_ret-100;                                   \
+        }                                                            \
+      if (0 > lwes_event_add_attr                                    \
+            (event, tmp_short_str, attr))                            \
+        {                                                            \
+          free(attr->value);                                         \
+          free(attr);                                                \
+          return set_fail_ret;                                       \
+        }                                                            \
+    }                                                                \
+  else if (tmp_byte == LWES_TYPE_N_##typ##_ARRAY)                    \
+    {                                                                \
+      struct lwes_event_attribute* attr;                             \
+      attr = lwes_event_attribute_create(tmp_byte, NULL, 0);         \
+      if (!unmarshall_array_attribute                                \
+            (attr, bytes, num_bytes, &tmpOffset))                    \
+        {                                                            \
+          lwes_event_attribute_destroy(attr);                        \
+          return mar_fail_ret-100;                                   \
```

Comment:
These also match between the two clauses, should they be different?

Reply:
Not sure why this preview still shows -100.. it's changed to -150.
Problems with large diffs, maybe?